apps/testingaccessibility/src/pages/learn/index.tsx

```diff
@@ -62,17 +62,26 @@ const productQuery = groq`*[_type == "product" && productId == $productId][0]{
   }`
 
 export const getServerSideProps: GetServerSideProps = async ({req}) => {
-  const {product, purchases, token} = await getPurchasedProduct(
-    req,
-    productQuery,
-  )
-  const ability = await getAbilityFromToken(token)
+  const session = await getSession({req})
+
+  // TODO: instead of fetching the product we can generate rules on the session
+  //       which may mean we can avoid this async call but it also isn't hurting
+  //       anything right now. The ability isn't being used to make any decisions
+  //       at this point.
+  const {product, purchases} = await getPurchasedProduct(req, productQuery)
+
+  const ability = getCurrentAbility({rules: session?.rules})
```

Comment:
technically in this case we could generate the rules based on the purchases above, but instead we'll just send in the cached rules so we aren't doing the "work" again

```
const ability = getCurrentAbility({purchases})
```

but in most circumstances we wouldn't be loading all purchases so just using rules here

**you can generate the ability using either rules or purchases**